pub/lib/dependabot/pub/update_checker.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require "dependabot/update_checkers"
+require "dependabot/update_checkers/base"
+require "dependabot/pub/helpers"
+
+module Dependabot
+  module Pub
+    class UpdateChecker < Dependabot::UpdateCheckers::Base
+      include Dependabot::Pub::Helpers
+
+      def initialize(dependency:, dependency_files:, repo_contents_path: nil,
+                     credentials:, ignored_versions: [],
+                     raise_on_ignored: false, security_advisories: [],
+                     requirements_update_strategy: nil,
+                     options: {}, pub_hosted_url: nil)
+        @pub_hosted_url = pub_hosted_url
```

Comment:
This is a bit of a nit, but could we fold the `pub_hosted_url` into the `options` hash? That way we don't need to alter the initializer for this class:

```suggestion
                     options: {})
        @pub_hosted_url = options.fetch(:pub_hosted_url, nil)
```

Reply:
I was considering that, but sure I was "allowed"

Done